src/main/java/org/bricolages/streaming/filter/TextOp.java

```diff
@@ -65,4 +65,15 @@ String castStringForce(Object value) {
         if (!(value instanceof String)) return null;
         return (String)value;
     }
+
+    final Pattern nullCharsPattern = Pattern.compile("^\\x00+$");
+    String removeHeadNullChars(String str) {
+        if (str == null) return null;
+        // return if first char is not \0
+        if (0 != str.indexOf("\0")) return str;
```

Comment:
Flipped condition expression (invert of `str.indexOf("\0") != 0`) is a kind of superstition.  Never use that.